log/klogv2/klogv2.go

```diff
@@ -96,3 +112,22 @@ func (l *adapter) WithFields(fields log.Fields) log.Logger {
 func (l *adapter) WithError(err error) log.Logger {
 	return l.WithField("err", err)
 }
+
+// processFields prepares a string to be appended to every log call.
+// This is called once to avoid costly log operations.
+func processFields(fields map[string]interface{}) string {
+	// Order fields lexically for easier reading.
+	keys := make([]string, 0, len(fields))
+	for k := range fields {
+		keys = append(keys, k)
```

Comment:
A heap would be more efficient.

Reply:
How much more efficient? Also, don't forget this is processed only once, when setting up the logger?

Reply:
ah, for some reason I thought this would get eval'd on each log call. scratch that.